### PR TITLE
Adding test guide for e2e suite

### PR DESCRIPTION
**Pull request by JaydipGabani**
This PR is dependent on #1153.

### CODE REVIEW

**Comment by djwhatle** [comment]
Let's link to this guide from somewhere in https://github.com/konveyor/mig-controller/blob/master/HACKING.md

**Comment by JaydipGabani** [comment]
> Let's link to this guide from somewhere in https://github.com/konveyor/mig-controller/blob/master/HACKING.md

Added the link in HACKING.md

File: docs/testguide.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Testing guide
+
+The purpose of this document is to provide start guide to set up, run and extend e2e test suite.
+
+### Prerequisites/Dependencies
+
+- Install Ginkgo and Gomega 
+
+      go get github.com/onsi/ginkgo/ginkgo
+      go get github.com/onsi/gomega/...
```
Comment (djwhatle) [inline docs/testguide.md:10]: @JaydipGabani what does the `...` signify here? Is that the actual command the user needs to run?

File: docs/testguide.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+```
+# this command should be run from root of the mig-controller repo
+make e2e-test 
+```
+
+### Structure of the current suite
+
+#### Set-up for the test suite
+`BeforeSuite` of `test_suite_test.go` is responsible for setting up the environment to the point where all the dependencies to create a plan and run a migration. This includes installing controller if not present, creation of `migStorage` and `migCluster` and making sure these are in the desired state. Currently `migSorage` is limited to `aws s3` bucket, any other type of `migStorage` creation will be added in future.
```
Comment (djwhatle) [inline docs/testguide.md:50]:
```suggestion
`BeforeSuite` of `test_suite_test.go` is responsible for setting up E2E test dependencies needed prior to creating a plan and running a migration. This ensures the following are present and in the desired state:
- `MigrationController`
- `MigStorage` 
- `MigCluster`

Currently `MigStorage` is limited to `aws s3` bucket. Support for other types of `MigStorage` creation will be added in future.   
```

**Review by djwhatle — CHANGES_REQUESTED** [review]
Added some suggested changes. Thanks for writing this up, definitely helpful!

File: docs/testguide.md (hunk 1 above)
Comment (JaydipGabani) [reply docs/testguide.md:10]: @djwhatle Yes!! I was also confused, but that is how they mentioned it on the documentation.

**Review by alaypatel07 — COMMENTED** [review]
LGTM

**Review by alaypatel07 — APPROVED** [review]

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]

**Review by pranavgaikwad — APPROVED** [review]
Thanks for arranging the variables in a table. LGTM

**Pull request merged**